src/x86/avx2.rs

```diff
@@ -568,24 +566,326 @@ pub fn _mm256_sad_epu8 (a: u8x32, b: u8x32) -> u64x4 {
+/// shifting in zeros, and return the result
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm256_sll_epi64(a: i64x4, count: i64x2) -> i64x4 {
+    unsafe { psllq(a, count) }
+}
+
+/// Shift packed 16-bit integers in `a` left by `imm8` while
+/// shifting in zeros, return the results;
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm256_slli_epi16(a: i16x16, imm8: i32) -> i16x16 {
+    unsafe { pslliw(a, imm8) }    
+}
+
+/// Shift packed 32-bit integers in `a` left by `imm8` while
+/// shifting in zeros, return the results;
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm256_slli_epi32(a: i32x8, imm8: i32) -> i32x8 {
+    unsafe { psllid(a, imm8) }    
+}
+
+/// Shift packed 64-bit integers in `a` left by `imm8` while
+/// shifting in zeros, return the results;
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm256_slli_epi64(a: i64x4, imm8: i32) -> i64x4 {
+    unsafe { pslliq(a, imm8) }    
+}
+
+// TODO _mm256_slli_si256 (__m256i a, const int imm8)
+
+/// Shift packed 32-bit integers in `a` left by the amount
+/// specified by the corresponding element in `count` while
+/// shifting in zeros, and return the result.
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm_sllv_epi32(a: i32x4, count: i32x4) -> i32x4 {
+    unsafe { psllvd(a, count) }
+}
+
+/// Shift packed 32-bit integers in `a` left by the amount
+/// specified by the corresponding element in `count` while
+/// shifting in zeros, and return the result.
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm256_sllv_epi32(a: i32x8, count: i32x8) -> i32x8 {
+    unsafe { psllvd256(a, count) }
+}
+
+/// Shift packed 64-bit integers in `a` left by the amount
+/// specified by the corresponding element in `count` while
+/// shifting in zeros, and return the result.
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm_sllv_epi64(a: i64x2, count: i64x2) -> i64x2 {
+    unsafe { psllvq(a, count) }
+}
+
+/// Shift packed 64-bit integers in `a` left by the amount
+/// specified by the corresponding element in `count` while
+/// shifting in zeros, and return the result.
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm256_sllv_epi64(a: i64x4, count: i64x4) -> i64x4 {
+    unsafe { psllvq256(a, count) }
+}
+
+/// Shift packed 16-bit integers in `a` right by `count` while
+/// shifting in sign bits.
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm256_sra_epi16(a: i16x16, count: i16x8) -> i16x16 {
+    unsafe { psraw(a, count) }
+}
+
+/// Shift packed 32-bit integers in `a` right by `count` while
+/// shifting in sign bits.
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm256_sra_epi32(a: i32x8, count: i32x4) -> i32x8 {
+    unsafe { psrad(a, count) }
+}
+
+/// Shift packed 16-bit integers in `a` right by `imm8` while 
+/// shifting in sign bits.
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm256_srai_epi16(a: i16x16, imm8: i32) -> i16x16 {
+    unsafe { psraiw(a, imm8) }
```

Comment:
Right. :-) It requires investigative work unfortunately. The fact that it works in debug mode now though is promising... but still worth double checking. I'll do it myself eventually, but not while I'm at work.